README.md

```diff
@@ -1,129 +1,75 @@
+2. CD into the generated app directory: `cd <APP_NAME>`
+3. Install the app's NPM dependencies, which includes this plugin: `npm install`
+4. Update the `service` property in your `serverless.yml` file, to ensure it specifies a unique app name:
 
-1. Set up an Azure Subscription
-
-    Sign up for a free account @ [https://azure.com](https://azure.microsoft.com/en-us/services/functions/).
-
-    Azure comes with a [free trial](https://azure.microsoft.com/en-us/free/) that includes $200 of free credit. 
-
-
-2. . Get the Azure CLI
-
-    ```
-    npm i -g azure-cli
-    ```
-
-3. Login to Azure
-
-    ```
-    azure login
-    ```
-
-    This will give you a code and prompt you to visit [aka.ms/devicelogin](https://aka.ms/devicelogin). Provide the code and then login with your Azure identity (this may happen automatically if you're already logged in). You'll then be able to access your account via the CLI.
-
-4. Get your subcription and tenant id
-
-    ```
-    azure account show
+    ```yml
+    service: my-azure-functions-app # Name of the Azure function App you want to create
     ```
 
-    Save the subcription and tenant id for later
+### Deploy, test, and diagnose your Azure service
 
-5. Create a service principal for a given `<name>` and `<password>` and add contributor role.
+1. Deploy your new service to Azure! The first time you do this, you will be asked to authenticate with your Azure account, so the `serverless` CLI can manage Functions on your behalf. Simply follow the provided instructions, and the deployment will continue as soon as the authentication process is completed.
 
-    ```
-    azure ad sp create -n <name> -p <password>
+    ```bash
+    serverless deploy
     ```
 
-    This should return an object which has the `servicePrincipalNames` property on it and an ObjectId. Save the Object Id and one of the names in the array and the password you provided for later. If you need to look up your service principal later, you can use `azure ad sp -c <name>` where `<name>` is the name provided originally. Note that the `<name>` you provided is not the name you'll provide later, it is a name in the `servicePrincipalNames` array.
+    > Note: Once you've authenticated, a new Azure "service principal" will be created, and used for subsequent deployments. This prevents you from needing to manually login again. See [below](#advanced-authentication) if you'd prefer to use a custom service principal instead.
 
-    Then grant the SP contributor access with the ObjectId
+2. Invoke a function, in order to test that it works:
 
     ```bash
-    azure role assignment create --objectId <objectIDFromCreateStep> -o Contributor
+    serverless invoke -f hello
     ```
 
-6. Set up environment variables
-
-     You need to set up environment variables for your subscription id, tenant id, service principal name, and password. 
+3. View the output logs of your newly deployed function:
```

Comment:
Change to : Stream the Function Logs. We have an [issue ](https://github.com/serverless/serverless-azure-functions/issues/8)tracking fetching logs for an invocation.